sources/updater/update_metadata/src/lib.rs

```diff
@@ -387,7 +337,41 @@ impl Update {
     }
 }
 
-pub fn migration_targets(from: &Version, to: &Version, manifest: &Manifest) -> Result<Vec<String>> {
+pub fn find_migrations(from: &Version, to: &Version, manifest: &Manifest) -> Result<Vec<String>> {
+    // early exit if there is no work to do.
+    if from == to {
+        return Ok(Vec::new());
+    }
+    // determine if the migration direction is forward or backward.
+    let direction = Direction::from_versions(from, to).unwrap_or(Direction::Forward);
+    // express the versions in ascending order
+    let (lower_version, higher_version) = order_versions(from, to);
+    let mut migrations = find_migrations_forward(&lower_version, &higher_version, manifest)?;
+    // if the direction is backward, reverse the order of the migration list.
+    if direction == Direction::Backward {
+        migrations = migrations.into_iter().rev().collect();
+    }
+    Ok(migrations)
+}
+
+// returns the versions with the lower version first.
+fn order_versions(a: &Version, b: &Version) -> (Version, Version) {
+    if let Some(direction) = Direction::from_versions(&a, &b) {
+        if direction == Direction::Backward {
+            return (b.clone(), a.clone());
+        }
+    }
+    (a.clone(), b.clone())
+}
```

Comment:
I made this simplification. I'm going to resolve conversations like these as I commit the fixes on my side branch (so that I know what I've addressed and what I haven't). I will aggregate several before doing a force push to the PR branch. You can wait for my re-review signal to avoid churn.